src/Microsoft.DotNet.ProjectModel/CompilationOutputFiles.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel.Resources;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class CompilationOutputFiles
+    {
+        protected readonly Project Project;
+        protected readonly string Configuration;
+        protected readonly NuGetFramework Framework;
+
+        public CompilationOutputFiles(
+            string basePath,
+            Project project,
+            string configuration,
+            NuGetFramework framework)
+        {
+            BasePath = basePath;
+            Project = project;
+            Configuration = configuration;
+            Framework = framework;
+
+            OutputExtension = FileNameSuffixes.DotNet.DynamicLib;
+
+            var compilationOptions = Project.GetCompilerOptions(framework, configuration);
+            if (framework.IsDesktop() && compilationOptions.EmitEntryPoint.GetValueOrDefault())
+            {
+                OutputExtension = FileNameSuffixes.DotNet.Exe;
+            }
+        }
+
+        public string BasePath { get; }
```

Comment:
Let's keep the properties declaration together? Isn't this the style convention? It should be.